src/lock/session.go

```diff
@@ -19,35 +23,38 @@ type Runner struct {
 	sessionMutex sync.RWMutex
 	sessionID    int64
 	dbFinder     DBFinder
+	client       metrics.Client
 	tracer       Tracer
 	scanTask     ScanTask
 	loopTick     time.Duration
 	logger       Logger
 	tasker       Tasker
+	name         string
 }
 
 // NewRunner will create a new Runner to handle a type of task
 // dbFinder can get an instance of the Database interface on demand
 // scanTask can read from a sql.row into a Task
 // tasker can complete Tasks
 // looptick defines how often to check for tasks to complete
-// tracer is optional and will log tracing metrics in an open tracing friendly way if provided
+// client is a go-metrics-client that will also start spans for us
 // logger is optional and will log errors if provided
-func NewRunner(dbFinder DBFinder, scanTask ScanTask, tasker Tasker, loopTick time.Duration, tracer Tracer, logger Logger) *Runner {
-	if tracer == nil {
-		tracer = newNoopTracer()
+func NewRunner(dbFinder DBFinder, scanTask ScanTask, tasker Tasker, loopTick time.Duration, client metrics.Client, logger Logger, name string) *Runner {
```

Comment:
can we keep using the interface so we dont have to import go-metric-client here?